### PR TITLE
Fix flaky netty executor instrumentation test

### PR DESCRIPTION
**Pull request by richardstartin**

### CODE REVIEW

**Review by mcculls — APPROVED** [review]

**Review by bantonsson — APPROVED** [review]
👍

Flaky tests are still flaky though https://app.circleci.com/pipelines/github/DataDog/dd-trace-java/3900/workflows/1ccd07e1-e20a-4738-a413-10384557ad68/jobs/100539

**Comment by richardstartin** [comment]
@bantonsson I changed the `RunnableFutureInstrumentation` and brought EPoll back in since it seems to be more stable (I could reproduce failure after a few minutes of running the tests before, now I can't).

**Comment by richardstartin** [comment]
It turns out the flakiness was caused by changing `doExec` (which is called during work stealing) to `exec`. The solution is to instrument both methods idempotently.

**Review by tylerbenson — APPROVED** [review]

File: dd-java-agent/instrumentation/java-concurrent/src/main/java/datadog/trace/instrumentation/java/concurrent/JavaForkJoinTaskInstrumentation.java [hunk 1]
```diff
@@ -58,7 +59,7 @@ public boolean matches(TypeDescription target) {
   @Override
   public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
     Map<ElementMatcher<MethodDescription>, String> transformers = new HashMap<>(4);
-    transformers.put(isMethod().and(named("exec")), getClass().getName() + "$Exec");
+    transformers.put(isMethod().and(namedOneOf("doExec", "exec")), getClass().getName() + "$Exec");
```
Comment (bantonsson) [inline dd-java-agent/instrumentation/java-concurrent/src/main/java/datadog/trace/instrumentation/java/concurrent/JavaForkJoinTaskInstrumentation.java:62]: 👍

**Review by bantonsson — APPROVED** [review]

File: dd-java-agent/instrumentation/java-concurrent/src/main/java/datadog/trace/instrumentation/java/concurrent/JavaForkJoinTaskInstrumentation.java (hunk 1 above)
Comment (richardstartin) [reply dd-java-agent/instrumentation/java-concurrent/src/main/java/datadog/trace/instrumentation/java/concurrent/JavaForkJoinTaskInstrumentation.java]: I changed this some more to revert the change made in #1971 _when work stealing takes place_. This isn't a reflection on that change but the weird semantics of that flag.

File: dd-java-agent/instrumentation/akka-concurrent/src/main/java/datadog/trace/instrumentation/akka/concurrent/AkkaForkJoinTaskInstrumentation.java [hunk 2]
```diff
@@ -44,6 +52,22 @@ public AkkaForkJoinTaskInstrumentation() {
     return named("akka.dispatch.forkjoin.ForkJoinTask");
   }
 
+  @Override
+  public boolean isEnabled() {
+    return true;
+  }
```
Comment (tylerbenson) [inline dd-java-agent/instrumentation/akka-concurrent/src/main/java/datadog/trace/instrumentation/akka/concurrent/AkkaForkJoinTaskInstrumentation.java:58]: This shouldn't be needed...

File: dd-java-agent/instrumentation/java-concurrent/src/main/java/datadog/trace/instrumentation/java/concurrent/JavaForkJoinTaskInstrumentation.java [hunk 3]
```diff
@@ -58,12 +65,28 @@ public boolean matches(TypeDescription target) {
...
     transformers.put(isMethod().and(named("fork")), getClass().getName() + "$Fork");
     transformers.put(isMethod().and(named("cancel")), getClass().getName() + "$Cancel");
     return Collections.unmodifiableMap(transformers);
   }
 
+  @Override
+  public boolean isEnabled() {
+    return true;
+  }
```
Comment (tylerbenson) [inline dd-java-agent/instrumentation/java-concurrent/src/main/java/datadog/trace/instrumentation/java/concurrent/JavaForkJoinTaskInstrumentation.java:77]: or this

**Review by tylerbenson — APPROVED** [review]

File: dd-java-agent/instrumentation/akka-concurrent/src/main/java/datadog/trace/instrumentation/akka/concurrent/AkkaForkJoinTaskInstrumentation.java (hunk 2 above)
Comment (richardstartin) [reply dd-java-agent/instrumentation/akka-concurrent/src/main/java/datadog/trace/instrumentation/akka/concurrent/AkkaForkJoinTaskInstrumentation.java:58]: This enables the filter provider

File: dd-java-agent/instrumentation/java-concurrent/src/main/java/datadog/trace/instrumentation/java/concurrent/JavaForkJoinTaskInstrumentation.java (hunk 3 above)
Comment (richardstartin) [reply dd-java-agent/instrumentation/java-concurrent/src/main/java/datadog/trace/instrumentation/java/concurrent/JavaForkJoinTaskInstrumentation.java:77]: and this

File: dd-java-agent/instrumentation/akka-concurrent/src/main/java/datadog/trace/instrumentation/akka/concurrent/AkkaForkJoinTaskInstrumentation.java [hunk 4]
```diff
@@ -44,6 +52,22 @@ public AkkaForkJoinTaskInstrumentation() {
...
+
```
Comment (randomanderson) [inline dd-java-agent/instrumentation/akka-concurrent/src/main/java/datadog/trace/instrumentation/akka/concurrent/AkkaForkJoinTaskInstrumentation.java:59]: Is there are reason we don't want enabling/disabling to work like every other instrumentation?

Comment (richardstartin) [reply dd-java-agent/instrumentation/akka-concurrent/src/main/java/datadog/trace/instrumentation/akka/concurrent/AkkaForkJoinTaskInstrumentation.java:59]: This is a misleading named method from `ExcludeFilterProvider` - I blame @bantonsson. Perhaps we should call it `isFilterEnabled`. It means that the filter returned by `excludedClasses` should be respected.

Comment (randomanderson) [reply dd-java-agent/instrumentation/akka-concurrent/src/main/java/datadog/trace/instrumentation/akka/concurrent/AkkaForkJoinTaskInstrumentation.java:59]: NVM.  Name confusion

File: dd-java-agent/instrumentation/akka-concurrent/src/main/java/datadog/trace/instrumentation/akka/concurrent/AkkaForkJoinTaskInstrumentation.java (hunk 2 above)
Comment (tylerbenson) [reply dd-java-agent/instrumentation/akka-concurrent/src/main/java/datadog/trace/instrumentation/akka/concurrent/AkkaForkJoinTaskInstrumentation.java:58]: shoot... so confusing. sorry about that.

Comment (bantonsson) [reply dd-java-agent/instrumentation/akka-concurrent/src/main/java/datadog/trace/instrumentation/akka/concurrent/AkkaForkJoinTaskInstrumentation.java:58]: So the reason this was there in the first place was to be able to have different behavior depending on if the instrumentation was enabled or not (in effect reverting to the old behavior for `CompletableFuture`), but it seems that we are way beyond that point anyway, so I'll just remove this method and the corresponding check altogether.

**Pull request merged**